openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSPatientServiceImpl.java

```diff
@@ -245,6 +253,56 @@ private Patient updatePatient(Config config, Patient patient) {
         return updatedPatient;
     }
 
+    private Patient updatePatientIdentifiers(Config config, Patient patient) {
+        Validate.notNull(patient, "Patient cannot be null");
+        Validate.notEmpty(patient.getUuid(), "Patient Id cannot be empty");
+
+        List<Identifier> identifiersList = patientResource.getPatientIdentifierList(config, patient.getUuid());
+        Patient updatedPatient;
+        try {
+            //Patient Identifiers have to be update separately.
+            updatePatientIdentifiers(config, patient, identifiersList);
+            updatedPatient = getPatientByUuid(config, patient.getUuid());
+            eventRelay.sendEventMessage(new MotechEvent(EventKeys.UPDATED_PATIENT_IDENTIFIERS_SUBJECT, EventHelper.patientParameters(updatedPatient)));
+        } catch (HttpClientErrorException e) {
+            throw new OpenMRSException("Failed to update OpenMRS patient's identifiers for patient with id: " + patient.getUuid(), e);
+        }
+        return updatedPatient;
+    }
+
+    private void updatePatientIdentifiers(Config config, Patient patient, List<Identifier> fetchedIdentifierList) {
+        String fetchedIdentifierName;
+        String newIdentifierName;
+
+
+        for (Identifier newIdentifier : patient.getIdentifiers()) {
+            boolean isUpdate = false;
+
+            for (Identifier fetchedIdentifier : fetchedIdentifierList) {
+                fetchedIdentifierName = fetchedIdentifier.getIdentifierType().getName();
+                newIdentifierName = newIdentifier.getIdentifierType().getName();
+
+                if (newIdentifierName.equals(fetchedIdentifierName)) {
+                    fetchedIdentifier.setIdentifier(newIdentifier.getIdentifier());
+                    patientResource.updatePatientIdentifier(config, patient.getUuid(), fetchedIdentifier);
+                    isUpdate = true;
+                }
+            }
+            if (!isUpdate) {
+                addNewPatientIdentifier(config, patient, newIdentifier);
+            }
+        }
+    }
+
+    private void addNewPatientIdentifier(Config config, Patient patient, Identifier newIdentifier) {
+        String uuid = patientResource.getPatientIdentifierTypeUuidByName(config, newIdentifier.getIdentifierType().getName());
```

Comment:
remove adding location, now only motech id needs location. The OpenMRS module doesn't support additional identifiers with location required.